pkg/kritis/cryptolib/verify_detached_test.go

```diff
@@ -0,0 +1,256 @@
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaSignPkcs14096Sha256 signature",
+			signature:     []byte(rsa4096_256Sig),
+			pubkey:        []byte(rsa4096PubKey),
+			signingAlg:    RsaSignPkcs14096Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaSignPkcs14096Sha512 signature",
+			signature:     []byte(rsa4096_512Sig),
+			pubkey:        []byte(rsa4096PubKey),
+			signingAlg:    RsaSignPkcs14096Sha512,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss2048Sha256 signature",
+			signature:     []byte(rsa2048_256PssSig),
+			pubkey:        []byte(rsa2048PubKey),
+			signingAlg:    RsaPss2048Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss3072Sha256 signature",
+			signature:     []byte(rsa3072_256PssSig),
+			pubkey:        []byte(rsa3072PubKey),
+			signingAlg:    RsaPss3072Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss4096Sha256 signature",
+			signature:     []byte(rsa4096_256PssSig),
+			pubkey:        []byte(rsa4096PubKey),
+			signingAlg:    RsaPss4096Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss4096Sha512 signature",
+			signature:     []byte(rsa4096_512PssSig),
+			pubkey:        []byte(rsa4096PubKey),
+			signingAlg:    RsaPss4096Sha512,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP256Sha256 signature",
+			signature:     []byte(ec256Sig),
+			pubkey:        []byte(ec256PubKey),
+			signingAlg:    EcdsaP256Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP384Sha384 signature",
+			signature:     []byte(ec384Sig),
+			pubkey:        []byte(ec384PubKey),
+			signingAlg:    EcdsaP384Sha384,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP521Sha512 signature",
+			signature:     []byte(ec521Sig),
+			pubkey:        []byte(ec521PubKey),
+			signingAlg:    EcdsaP521Sha512,
+			payload:       []byte(goodPayload),
+			expectedError: false,
+		},
+		{
+			name:          "bad pub key",
+			signature:     []byte(rsa2048_256Sig),
+			pubkey:        []byte(badKey),
+			signingAlg:    RsaSignPkcs12048Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: true,
+		},
+		{
+			name:          "pub key with extra data",
+			signature:     []byte(rsa2048_256Sig),
+			pubkey:        []byte(extraDataKey),
+			signingAlg:    RsaSignPkcs12048Sha256,
+			payload:       []byte(goodPayload),
+			expectedError: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			decoded_sig, decodeErr := base64.RawURLEncoding.DecodeString(string(tc.signature))
```

Comment:
Since this is the only place `tc.signature` is used, you can define it as a `string` in the test struct. Then you don't have to cast it here.

Reply:
done.